File: src/main/java/com/epicdragonworld/util/ConfigReader.java
package com.epicdragonworld.util;

import java.io.FileInputStream;
import java.util.Properties;
import java.util.logging.Logger;

/**
 * @author Pantelis Andrianakis
 */
public final class ConfigReader
{
	private static final Logger LOGGER = Logger.getLogger(ConfigReader.class.getName());
	
	private final Properties _properties = new Properties();
	private final String _fileName;
	
	public ConfigReader(String fileName)
	{
		_fileName = fileName;
		try
		{
			_properties.load(new FileInputStream(fileName));
		}
		catch (Exception e)
		{
			LOGGER.warning("Could not load " + fileName + ". " + e.getMessage());
		}
	}
	
	public String getString(String config, String defaultValue)
	{
		if (!_properties.containsKey(config))
		{
			LOGGER.warning("Missing config " + config + " from file " + _fileName + ". Default value " + defaultValue + " will be used instead.");
			return defaultValue;
		}
		return _properties.getProperty(config);
	}
	
	public boolean getBoolean(String config, boolean defaultValue)
	{
		if (!_properties.containsKey(config))
		{
			LOGGER.warning("Missing config " + config + " from file " + _fileName + ". Default value " + defaultValue + " will be used instead.");
			return defaultValue;
		}
		
		try
		{
			return Boolean.parseBoolean(_properties.getProperty(config));
		}
		catch (Exception e)
		{
			LOGGER.warning("Config " + config + " from file " + _fileName + " should be Boolean. Using default value " + defaultValue + " instead.");
			return defaultValue;
		}
	}
	
	public int getInt(String config, int defaultValue)
	{
		if (!_properties.containsKey(config))
		{
			LOGGER.warning("Missing config " + config + " from file " + _fileName + ". Default value " + defaultValue + " will be used instead.");
			return defaultValue;
		}
		
		try
		{
			return Integer.parseInt(_properties.getProperty(config));
		}
		catch (Exception e)
		{
			LOGGER.warning("Config " + config + " from file " + _fileName + " should be Integer. Using default value " + defaultValue + " instead.");
			return defaultValue;
		}
	}
	
	public long getLong(String config, long defaultValue)
	{
		if (!_properties.containsKey(config))
		{
			LOGGER.warning("Missing config " + config + " from file " + _fileName + ". Default value " + defaultValue + " will be used instead.");
			return defaultValue;
		}
		
		try
		{
			return Long.parseLong(_properties.getProperty(config));
		}
		catch (Exception e)
		{
			LOGGER.warning("Config " + config + " from file " + _fileName + " should be Long. Using default value " + defaultValue + " instead.");
			return defaultValue;
		}
	}
	
	public float getFloat(String config, float defaultValue)
	{
		if (!_properties.containsKey(config))
		{
			LOGGER.warning("Missing config " + config + " from file " + _fileName + ". Default value " + defaultValue + " will be used instead.");
			return defaultValue;
		}
		
		try
		{
			return Float.parseFloat(_properties.getProperty(config));
		}
		catch (Exception e)
		{
			LOGGER.warning("Config " + config + " from file " + _fileName + " should be Float. Using default value " + defaultValue + " instead.");
			return defaultValue;
		}
	}
	
	public double getDouble(String config, double defaultValue)
	{
		if (!_properties.containsKey(config))
		{
			LOGGER.warning("Missing config " + config + " from file " + _fileName + ". Default value " + defaultValue + " will be used instead.");
			return defaultValue;
		}
		
		try
		{
			return Double.parseDouble(_properties.getProperty(config));
		}
		catch (Exception e)
		{
			LOGGER.warning("Config " + config + " from file " + _fileName + " should be Double. Using default value " + defaultValue + " instead.");
			return defaultValue;
		}
	}
}

File: EpicDragonWorldServer/Util/ConfigReader.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.IO;
using System.Linq;

/**
 * Author: Pantelis Andrianakis
 * Date: November 7th 2018
 */
class ConfigReader
{
    readonly Dictionary<string, string> configs = new Dictionary<string, string>();
    readonly string fileName;

    public ConfigReader(string fileName)
    {
        this.fileName = fileName;
        try
        {
            foreach (string row in File.ReadAllLines(fileName))
            {
                if (!row.StartsWith("#") && row.Trim().Length > 0)
                {
                    string[] rowSplit = row.Split('=');
                    if (rowSplit.Length > 1)
                    {
                        configs.Add(rowSplit[0].Trim(), string.Join("=", rowSplit.Skip(1).ToArray()).Trim());
                    }
                }
            }
        }
        catch (Exception e)
        {
            LogManager.Log("Could not load " + fileName + ". " + e.Message);
        }
    }

    public string GetString(string config, string defaultValue)
    {
        if (!configs.ContainsKey(config))
        {
            LogManager.Log("Missing config " + config + " from file " + fileName + ". Default value " + defaultValue + " will be used instead.");
            return defaultValue;
        }
        return configs[config];
    }

    public bool GetBool(string config, bool defaultValue)
    {
        if (!configs.ContainsKey(config))
        {
            LogManager.Log("Missing config " + config + " from file " + fileName + ". Default value " + defaultValue + " will be used instead.");
            return defaultValue;
        }

        try
        {
            return bool.Parse(configs[config]);
        }
        catch (Exception)
        {
            LogManager.Log("Config " + config + " from file " + fileName + " should be Boolean. Using default value " + defaultValue + " instead.");
            return defaultValue;
        }
    }

    public int GetInt(string config, int defaultValue)
    {
        if (!configs.ContainsKey(config))
        {
            LogManager.Log("Missing config " + config + " from file " + fileName + ". Default value " + defaultValue + " will be used instead.");
            return defaultValue;
        }

        try
        {
            return int.Parse(configs[config]);
        }
        catch (Exception)
        {
            LogManager.Log("Config " + config + " from file " + fileName + " should be Integer. Using default value " + defaultValue + " instead.");
            return defaultValue;
        }
    }

    public long GetLong(string config, long defaultValue)
    {
        if (!configs.ContainsKey(config))
        {
            LogManager.Log("Missing config " + config + " from file " + fileName + ". Default value " + defaultValue + " will be used instead.");
            return defaultValue;
        }

        try
        {
            return long.Parse(configs[config]);
        }
        catch (Exception)
        {
            LogManager.Log("Config " + config + " from file " + fileName + " should be Long. Using default value " + defaultValue + " instead.");
            return defaultValue;
        }
    }

    public float GetFloat(string config, float defaultValue)
    {
        if (!configs.ContainsKey(config))
        {
            LogManager.Log("Missing config " + config + " from file " + fileName + ". Default value " + defaultValue + " will be used instead.");
            return defaultValue;
        }

        try
        {
            return float.Parse(configs[config], CultureInfo.InvariantCulture);
        }
        catch (Exception)
        {
            LogManager.Log("Config " + config + " from file " + fileName + " should be Float. Using default value " + defaultValue + " instead.");
            return defaultValue;
        }
    }

    public double GetDouble(string config, double defaultValue)
    {
        if (!configs.ContainsKey(config))
        {
            LogManager.Log("Missing config " + config + " from file " + fileName + ". Default value " + defaultValue + " will be used instead.");
            return defaultValue;
        }

        try
        {
            return double.Parse(configs[config], CultureInfo.InvariantCulture);
        }
        catch (Exception)
        {
            LogManager.Log("Config " + config + " from file " + fileName + " should be Double. Using default value " + defaultValue + " instead.");
            return defaultValue;
        }
    }
}
